### PR TITLE
feat (api): export application with encrypted secrets

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/api/project/dao_key.go [hunk 1]
```diff
@@ -56,3 +56,22 @@ func DeleteProjectKey(db gorp.SqlExecutor, projectID int64, keyName string) erro
 	_, err := db.Exec("DELETE FROM project_key WHERE project_id = $1 AND name = $2", projectID, keyName)
 	return sdk.WrapError(err, "DeleteProjectKey> Cannot delete key %s", keyName)
 }
+
+func loadBuildinKey(db gorp.SqlExecutor, projectID int64) (k sdk.ProjectKey, err error) {
```
Comment (bnjjj) [inline engine/api/project/dao_key.go]: why named return ?

File: sdk/exportentities/application.go [hunk 2]
```diff
@@ -69,242 +33,54 @@ func NewApplication(app *sdk.Application) (a *Application) {
 
 	a.Variables = make(map[string]VariableValue, len(app.Variable))
 	for _, v := range app.Variable {
+		fmt.Println(v)
```
Comment (bnjjj) [inline sdk/exportentities/application.go]: to delete

File: engine/api/project/dao_key.go (hunk 1 above)
Comment (fsamin) [reply engine/api/project/dao_key.go]: All return statements precise the returned values
Here it's usefull to init the returned values which are used everywhere

Comment (bnjjj) [reply engine/api/project/dao_key.go]: ok not really fan but ok, however at this line https://github.com/ovh/cds/pull/1543/files/690a39286457e62bcf846a7b73c1706fbe5a4e56#diff-7f65e53ab8b93430efdc845089d28284R62 you make some kind of variable shadowing

**Pull request merged**